Activities/Cryptography/UiPath.Cryptography.Activities/KeyedHashText.cs

```diff
@@ -75,6 +69,16 @@ protected override void CacheMetadata(CodeActivityMetadata metadata)
                 var error = new ValidationError(Resources.FipsComplianceWarning, true, nameof(Algorithm));
                 metadata.AddValidationError(error);
             }
+            if (Algorithm.ToString().StartsWith("HMAC") && Key == null && KeySecureString == null)
+            {
+                var error1 = new ValidationError(Resources.KeyAndSecureStringNull, false, nameof(Key));
```

Comment:
Not sure what this block does and why we need to add a blank validation error.

Reply:
This block adds the validation on the Key field, if the field is empty and the algorithm starts with "HMAC". The blank validation is used for having the validation shown on the KeySecureString field also, but without duplicating the message.